addons/website/tests/test_lang_url.py

```diff
@@ -28,40 +30,54 @@ def test_02_url_redirect(self):
         url = '/fr_WHATEVER/contactus'
         r = self.url_open(url)
         self.assertEqual(r.status_code, 200)
-        self.assertTrue(r.url.endswith('/fr/contactus'), "fr_WHATEVER should be forwarded to 'fr_FR' lang as closest match")
+        self.assertTrue(r.url.endswith('/fr/contactus'), f"fr_WHATEVER should be forwarded to 'fr_FR' lang as closest match, url: {r.url}")
 
         url = '/fr_FR/contactus'
         r = self.url_open(url)
         self.assertEqual(r.status_code, 200)
-        self.assertTrue(r.url.endswith('/fr/contactus'), "lang in url should use url_code ('fr' in this case)")
+        self.assertTrue(r.url.endswith('/fr/contactus'), f"lang in url should use url_code ('fr' in this case), url: {r.url}")
 
     def test_03_url_cook_lang_not_available(self):
         """ An activated res.lang should not be displayed in the frontend if not a website lang. """
         self.website.language_ids = self.env.ref('base.lang_en')
+        self.website.flush()
         r = self.url_open('/fr/contactus')
```

Comment:
Maybe we should automatically `flush()` in method `url_open`, like we do in method `browser_js`...

Reply:
Ooops, that's old code, `flush()` is not necessary. I'm removing it.